lib/src/lifecycle_module.dart

```diff
@@ -136,7 +182,55 @@ abstract class LifecycleModule {
     return completer.future;
   }
 
+  /// Public method to suspend the module.
+  ///
+  /// Suspend indicates to the module that it should go into a low-activity
+  /// state. For example, by disconnecting from backend services and unloading
+  /// heavy data structures.
+  Future suspend() {
+    Completer completer = new Completer();
+    if (!_isSuspended && _isLoaded) {
+      _willSuspendController.add(this);
+      Future.wait(_childModules.map((c) => c.suspend())).then((_) {
+        onSuspend().then((_) {
+          _didSuspendController.add(this);
+          _isSuspended = true;
+          completer.complete();
+        });
+      });
+    } else {
+      _logger.warning(
+          'Module "$name" is already suspended or not loaded, cannot suspend.');
```

Comment:
we have a way to check these conditions so why not use a message that's more specific?
```
_logger.warning('Module "$name" is ${_isLoaded ? 'already suspended' : 'not loaded'}, cannot suspend.');
```